sklearn/tests/test_naive_bayes.py

```diff
@@ -63,6 +63,27 @@ def test_gnb_prior():
     assert_array_almost_equal(clf.class_prior_.sum(), 1)
 
 
+def test_gnb_sample_weight():
+    """Test whether sample weights are properly used in GNB. """
+    # Sample weights all being 1 should not change results
+    sw = np.ones(6)
+    clf = GaussianNB().fit(X, y)
+    clf_sw = GaussianNB().fit(X, y, sw)
+
+    assert_array_almost_equal(clf.theta_, clf_sw.theta_)
+    assert_array_almost_equal(clf.sigma_, clf_sw.sigma_)
+
+    # Fitting twice with half sample-weights should result
+    # in same result as fitting once with full weights
+    sw = np.random.RandomState(42).rand(y.shape[0])
```

Comment:
I think I'd prefer a test where you explicitly construct a dataset with duplicate points, so that not all sample weights are the same.